### PR TITLE
AzureML tabular + file kedro datasets 

### PR DESCRIPTION
**Pull request by asafalinadsg**
This PR includes the following:
1. `file_dataset.py` and `pandas_dataset.py` python modules, which contain AzureML file and tabular datasets integration with kedro datasets, respectively. You can view the docstrings of both `AzureMLFileDataSet` and `AzureMLPandasDataSet` classes for more info and examples of how to use. These integrations are tested manually by me - looks functionally good.
2. Moving of `runner_dataset.py` module to `datasets` directory, so that all datasets are under the same directory. I didnt test the impact of the changes done, however the imports of this module should work as before (for example, under `runner.py` module, the import `from kedro_azureml.datasets import KedroAzureRunnerDataset, KedroAzureRunnerDistributedDataset` should still work as expected)
3. Addition of new dependency, `azureml-core` (azureml SDK v1), which is essential for the file + tabular datasets integrations.

### CODE REVIEW

File: kedro_azureml/datasets/file_dataset.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    blob_path: str
+
+    @classmethod
+    def from_abfs_path(cls, path: str):
+        """
+        Parse blob storage path, assuming path is in the format:
+        - abfs://<storage_account_name>.blob.core.windows.net/<storage_container>/<blob_path>
+        - abfs://<storage_container>/<blob_path>
+        """
+        p = path.replace('abfs://', '')
```
Comment (szczeles) [inline kedro_azureml/datasets/file_dataset.py:24]: Let's add some unit tests for this functionality (covering implemented cases)

**Comment by szczeles** [comment]
I appreciate great docstrings but consider also adding a small paragraph in the docs, describing both datasets and their params (or maybe we can generate these from in-code docs, @marrrcin ?)

**Review by marrrcin — COMMENTED** [review]
Nice work @asafalinadsg, thanks for you contribution!

As @szczeles mentioned:
* please prepare some unit tests for the testable parts
* Please mention in the docs that there are 2 datasets that can be used and just point a reference to them. We will need to change our setup for the auto-docs to work with RTD - let's leave it for now.

I've verified both datasets manually for both read and write and they seem to work! 🎉

Since the datasets themselves are difficult to unit test, due to lots of dependencies on the Azure SDK and permissions, I think that it will be great to add them to our e2e tests.
You could e.g. add an entry for
```
preprocessed_companies:
  type: kedro_azureml.datasets.pandas_dataset.AzureMLPandasDataSet
  azureml_dataset: e2e_tests_preprocessed_companies
```

in the `tests/conf/e2e/catalog.yml` and pass azure storage account key and name via env vars here https://github.com/getindata/kedro-azureml/blob/d309b90782430920f838dc1a2a16d98ac08d21e4/.github/workflows/tests_and_publish.yml#L178
(the variables will be set via our CI/CD after the merge).

Not sure about the `AzureMLFileDataSet`, since it's based on a `PartitionedDataSet`, which something that is not used in the spaceflights starter we use for e2e tests. Any ideas?

**Comment by marrrcin** [comment]
Please also make sure that run `pre-commit` before pushing the code, right now it fails in CI/CD.

**Comment by tomasvanpottelbergh** [comment]
I was working on another implementation, but I didn't know about `PartitionedDataSet`, which is very nice to avoid having to implement this using dynamic inheritance!

Is there any reason why using the V2 SDK is not possible? Given that the [fsspec for Azure ML URLs](https://learn.microsoft.com/en-us/azure/machine-learning/how-to-access-data-interactive?tabs=adls) has now been released, I think it would be possible to use that? I can test that out if you like.

**Comment by marrrcin** [comment]
@tomasvanpottelbergh it would be nice to check out the Azure ML fsspec, but from what I see in the docs it seems to me like it's read-only 🤔

**Comment by eliorc** [comment]
> * Please mention in the docs that there are 2 datasets that can be used and just point a reference to them. We will need to change our setup for the auto-docs to work with RTD - let's leave it for now.

@marrrcin I'll try to contribute here as well, for the fixing part
I'm not sure what should be changed, is is the format? Would you rather have this as ReStructuredText?

**Comment by eliorc** [comment]
> Please also make sure that run `pre-commit` before pushing the code, right now it fails in CI/CD.

Fixed the pre-commit issues for now

**Comment by marrrcin** [comment]
> > * Please mention in the docs that there are 2 datasets that can be used and just point a reference to them. We will need to change our setup for the auto-docs to work with RTD - let's leave it for now.
> 
> @marrrcin I'll try to contribute here as well, for the fixing part I'm not sure what should be changed, is is the format? Would you rather have this as ReStructuredText?

The change in our setup will be needed for the [sphinx's autodoc](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html) to work, but we will handle this on our own. The only thing I'm asking here is just to add a paragraph to the existing docs that will point out to the new datasets. You can as well copy-paste the docstrings you have prepared so far.

**Comment by marrrcin** [comment]
Almost done, just the docs are not displaying:
https://kedro-azureml--49.org.readthedocs.build/en/49/source/04_data_assets.html
Screenshot 2023-03-06 at 11 02 00

due to:
```
WARNING: autodoc: failed to import class 'datasets.AzureMLPandasDataSet' from module 'kedro_azureml'; the following exception was raised:
No module named 'pandas'
WARNING: autodoc: failed to import class 'datasets.AzureMLFileDataSet' from module 'kedro_azureml'; the following exception was raised:
No module named 'pandas'
```

Please add the following:
```
autodoc_mock_imports = ["azureml", "pandas"]
```

to the `docs/conf.py` - it should fix it :)

File: pyproject.toml [hunk 2]
```diff
@@ -42,6 +42,7 @@ pydantic = "~=1.9.1"
 mlflow = {version = "^1.27.0", optional = true}
 backoff = "^2.2.1"
 azure-core = ">=1.26.1"
+azureml-core = "^1.49.0"
```
[the remark this answers is not in the document]
Comment (szczeles) [reply pyproject.toml:45]: I also needed to add this library to Poetry in order to avoid the failures

**Comment by szczeles** [comment]
Released in version 0.3.6 of the plugin.

Thanks for your contributions! :heart:

**Pull request merged**